nnet_utils/nnet_activation.h

```diff
@@ -186,50 +186,47 @@ void  softmax(data_T data[CONFIG_T::n_in], res_T res[CONFIG_T::n_in])
 
     // Index into the lookup table based on data for exponentials
     typename CONFIG_T::table_t exp_res[CONFIG_T::n_in];// different, independent, fixed point precision
-    typename CONFIG_T::table_t exp_res_sum=0;
+    typename CONFIG_T::table_t exp_diff_res[CONFIG_T::n_in][CONFIG_T::n_in];// different, independent, fixed point precision
     int data_round;
     int index;
     for (int ii=0; ii<CONFIG_T::n_in; ii++) {
-        // #pragma HLS UNROLL
-        data_round = data[ii]*CONFIG_T::table_size/16;
-        index = data_round + 8*CONFIG_T::table_size/16;
-        if (index < 0)   index = 0;
-        if (index > CONFIG_T::table_size-1) index = CONFIG_T::table_size-1;
-        exp_res[ii] = exp_table[index];
-        exp_res_sum += exp_table[index];
+      exp_res[ii] = 0;
+      for (int jj=0; jj<CONFIG_T::n_in; jj++) {
+	if (ii==jj) exp_diff_res[ii][jj] = 1;
+	else {
+	  data_round = (data[ii]-data[jj])*CONFIG_T::table_size/16;
```

Comment:
Found one bug. This should be data[jj]-data[ii].

Reply:
ack you're right! is it fast to redo the scan with this fix? i'm checking quickly that this gives the same output for the test bench

Reply:
mmh, it takes about 3 hours. I already checked the results of the testbench for <18,8> and after the fix I get the same results as for the other softmax implementations. But there are other problems with the division which I am looking at now. For instance for <16,4> I see a lot of zeros that should not be there. i'll let you know what else I find out.

Reply:
ok if we fix that bug we should be mostly there...

`master` branch, example-prjs/conv1-layer:
```
0.527591 (expected 0.528625, -0.195696 percent difference)
0.351451 (expected 0.352447, -0.282627 percent difference)
0.119575 (expected 0.117572, 1.70421 percent difference)
0.000261545 (expected 0.000254869, 2.61927 percent difference)
8.61883e-05 (expected 8.45194e-05, 1.97461 percent difference)
0.0010345 (expected 0.00101596, 1.82458 percent difference)
```

`jmgd/softmax-invert-lut` branch (bug)
```
0 (expected 0.528625, -100 percent difference)
0.0234375 (expected 0.352447, -93.3501 percent difference)
0 (expected 0.117572, -100 percent difference)
0.238281 (expected 0.000254869, 93391.5 percent difference)
0.726563 (expected 8.45194e-05, 859540 percent difference)
0.0585938 (expected 0.00101596, 5667.32 percent difference)
```

`jmgd/softmax-invert-lut` branch + bug fix:
```
0.533203 (expected 0.528625, 0.865951 percent difference)
0.355469 (expected 0.352447, 0.857234 percent difference)
0.118164 (expected 0.117572, 0.503923 percent difference)
0.0273438 (expected 0.000254869, 10628.5 percent difference)
0.015625 (expected 8.45194e-05, 18386.9 percent difference)
0 (expected 0.00101596, -100 percent difference
```